core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2580,6 +2581,61 @@ private void loadMetadataIfNotExist(RpcContext rpcContext, LockedInodePath inode
     }
   }
 
+  @Override
+  public void updateMount(AlluxioURI alluxioPath, MountContext context)
+      throws FileAlreadyExistsException, FileDoesNotExistException, InvalidPathException,
+      IOException, AccessControlException {
+    LockingScheme lockingScheme = createLockingScheme(alluxioPath,
+        context.getOptions().getCommonOptions(), LockPattern.WRITE_EDGE);
+    try (RpcContext rpcContext = createRpcContext();
+         LockedInodePath inodePath = mInodeTree
+             .lockInodePath(lockingScheme.getPath(), lockingScheme.getPattern());
+         FileSystemMasterAuditContext auditContext = createAuditContext(
+             "updateMount", alluxioPath, null, inodePath.getParentInodeOrNull())) {
+      try {
+        mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
+      } catch (AccessControlException e) {
+        auditContext.setAllowed(false);
+        throw e;
+      }
+      MountInfo mountInfo = mMountTable.getMountTable().get(alluxioPath.getPath());
+      if (mountInfo == null) {
+        throw new InvalidPathException("Failed to update mount properties for "
+            + inodePath.getUri() + ". Please ensure the path is an existing mount point.");
+      }
+      // test the new mount configuration first before unmounting the old ufs
+      UnderFileSystemConfiguration ufsConf = UnderFileSystemConfiguration
+          .defaults(ServerConfiguration.global())
+          .createMountSpecificConf(context.getOptions().getPropertiesMap());
+      String ufsUri = mountInfo.getUfsUri().toString();
+      try (UnderFileSystem ufs = UnderFileSystem.Factory.create(ufsUri, ufsConf)) {
+        // check the root path is valid
+        ufs.getFileStatus(ufsUri);
+      } catch (Throwable t) {
+        throw new InvalidArgumentException("Failed to mount under file system."
+            + " Please ensure the options are valid", t);
+      }
+      unmountInternal(rpcContext, inodePath, false);
+      long mountId = IdUtils.createMountId();
+      try {
+        mountInternal(rpcContext, inodePath, mountInfo.getUfsUri(), mountId, context, true);
+      } catch (Throwable t) {
+        // With ignoreError = true we should not normally get an exception. At this point we failed
+        // to add the mount point back to the table so we need to remove all child inodes and
+        // nested mount.
+        try {
+          deleteInternal(rpcContext, inodePath, DeleteContext
+              .mergeFrom(DeletePOptions.newBuilder().setRecursive(true).setAlluxioOnly(true)));
```

Comment:
It is a bit counter intuitive if failing the update causes the mount the be unmounted

Reply:
yeah, that's why I added all the extra code to:

1. creating a UFS client to validate the new options.
2. disable most of the checks that can cause a mount to fail.

Then, if the mount operation still fails(this will be highly unlikely), we have to do some clean up to make sure that we don't have the file metadata existing for a non-exist mount point. So this is a second level fail-safe mechanism. 

Most of the time if user specifies some invalid options for the mount point they will get an error message and the UFS should still work as before and its options are not updated.